src/Compilers/Core/MSBuildTask/ManagedToolTask.cs

```diff
@@ -159,7 +159,7 @@ protected static ITaskItem[] GenerateCommandLineArgsTaskItems(List<string> comma
             var items = new ITaskItem[commandLineArgs.Count];
             for (var i = 0; i < commandLineArgs.Count; i++)
             {
-                items[i] = new TaskItem(commandLineArgs[i]);
+                items[i] = new ArgsTaskItem(commandLineArgs[i]);
```

Comment:
I had thought we were going to try some trick about cloning a custom implementation. Does something like this not work?

- We create an ArgsTaskItem that implements ITaskItem2 just to return [EvaluatedIncludeEscaped](https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.framework.itaskitem2.evaluatedincludeescaped?view=msbuild-17-netcore)
- We wrap then call new TaskItem() passing that task specifically to hit [this line](https://github.com/dotnet/msbuild/blob/37eb419ad2c986ac5530292e6ee08e962390249e/src/Utilities/TaskItem.cs#L132) that looks like it'll bypass further escaping.

That way we're returning an existing implementation rather than returning our own which risks bugs in case we don't get the semantics of the interface right.

Reply:
Yep, let's do that instead.